### PR TITLE
Visitor can view a collection of products

### PR DESCRIPTION
**Pull request by Fraseer**
PT Link: https://www.pivotaltracker.com/story/show/179716947

### User Story:
As a visitor,
In order to get an overview of the restaurant's products
I would like to be able to see a menu

### Changes Proposed
- Creates tests for the first item on the menu
- Adds data file to store menu items
- Adds useEffect & axios to draw data from an api
- Creates MenuItems component that creates the structure for the menu
- Displays menu items as a list on Menu Tab

### What we learned working on this feature
- map function
- 

### Packages / Gems added
- Axios

### Screenshots (Videos)
image

### CODE REVIEW

File: cypress/integration/userCanSeeTheMenuList.feature.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+describe("User can see menu", () => {
```
Comment (offcast-dev) [inline cypress/integration/userCanSeeTheMenuList.feature.js:1]: Ok so if you are testing the whole menu there should be some test that indicates that you actually map over items. I suggest give all items identical data-cy. Then when you select them `cy.get(['data-cy=menu-item'])` cypress will get the array of all items that you can check the length of it `.should('have.length', 6)`. How to access individual items in array check for yourself in documentation.

File: cypress/integration/userCanSeeTheMenuList.feature.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    cy.visit("/");
+    cy.get("[data-cy=menu]").click();
+  });
+
+  it("is expected to show the first item in the Starters menu", () => {
+    cy.get("[data-cy=item-1]").within(() => {
+      cy.get(".header").should("contain", "Insect")
+      cy.get(".description").should("contain", "Bouncy bouncy");
+      cy.get(".item").should("contain", "500 kr");
+      cy.get(".image").should("be.visible");
```
Comment (offcast-dev) [inline cypress/integration/userCanSeeTheMenuList.feature.js]: use data-cy

File: public/data/menuItems.json [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+  },
+  {
+    "id": 2,
+    "category": "dessert",
+    "dish": "Feet Shavings",
+    "description": "Smelly Belly",
+    "price": 100,
+    "url": "https://picsum.photos/200"
+  }
+]
```
Comment (offcast-dev) [inline public/data/menuItems.json]: I understand if you use a file just to see if it works, but you have to make real API call and use `cy.intercept` to stub it out with this JSON

File: src/components/Menu.jsx [hunk 4]
```diff
@@ -1,11 +1,29 @@
...
+import { Segment } from "semantic-ui-react";
+import MenuItems from "./MenuItems";
+import axios from "axios";
 
 const Menu = () => {
+  const [menuItems, setIdentification] = useState([]);
+  useEffect(() => {
+    axios.get("./data/menuItems.json").then((response) => {
+      setIdentification(response.data);
+    });
```
Comment (offcast-dev) [inline src/components/Menu.jsx]: You have to make real API call here

File: src/components/Menu.jsx [hunk 5]
```diff
@@ -1,11 +1,29 @@
...
+  }, []);
+
+  let menuList = menuItems.map((item) => {
+    return (
+      <div data-cy={`item-${item.id}`} key={item.id}>
+        <MenuItems item={item} />
```
Comment (offcast-dev) [inline src/components/Menu.jsx]: Why plural? It is one MenuItem

File: src/components/MenuItems.jsx [hunk 6]
```diff
@@ -0,0 +1,20 @@
+import React from "react";
+import { List, Image } from "semantic-ui-react";
+
+const MenuItems = ({ item }) => {
+  return (
+    <List horizontal>
```
Comment (offcast-dev) [inline src/components/MenuItems.jsx]: So now you are mapping Lists while you should map List.Items. I say take `<List horizontal>` out of this component so it wraps all List.Items, thus I don't think you need this one  `<div data-cy={`item-${item.id}`} key={item.id}>`.

**Review by offcast-dev — CHANGES_REQUESTED** [review]
Good, but there is room for improvement. Biggest deal breaker is that you don't do real API call

File: cypress/integration/userCanSeeTheMenuList.feature.js (hunk 1 above)
Comment (Fraseer) [reply cypress/integration/userCanSeeTheMenuList.feature.js:1]: Added test to check the number of items on the menu.

File: cypress/integration/userCanSeeTheMenuList.feature.js (hunk 2 above)
Comment (Fraseer) [reply cypress/integration/userCanSeeTheMenuList.feature.js]: Added data-cy to tests

File: src/components/Menu.jsx (hunk 5 above)
Comment (Fraseer) [reply src/components/Menu.jsx]: Changed

File: public/data/menuItems.json (hunk 3 above)
Comment (Fraseer) [reply public/data/menuItems.json]: Now intercepting and successfully stubbing

File: src/components/Menu.jsx (hunk 4 above)
Comment (Fraseer) [reply src/components/Menu.jsx]: As far as we can tell axios is now making call, based off the network call in Fetch/XHR

Comment (tochman) [reply src/components/Menu.jsx]: Yes, but you need to make the call to the API the back-end ppl are building, not to a file in your own folder structure.

File: src/components/MenuItems.jsx (hunk 6 above)
Comment (Fraseer) [reply src/components/MenuItems.jsx]: We have moved <List horizontal> out of the MenuItem Component, and removed key={item.id}, but we think we need data-cy={item-${item.id}} to be able to choose which item we run tests on.

**Comment by Fraseer** [comment]
I have push the changes we have mentioned above

File: src/components/Menu.jsx [hunk 7]
```diff
@@ -1,11 +1,29 @@
...
 
 const Menu = () => {
+  const [menuItems, setIdentification] = useState([]);
+  useEffect(() => {
+    axios.get("https://slowfood.heroku.com/api/products").then((response) => {
+      setIdentification(response.data.products);
+    });
+  }, []);
+
+  let menuList = menuItems.map((items) => {
```
Comment (tochman) [inline src/components/Menu.jsx]: When you iterate over this array, wouldn't it be better to use the name `item` vs `items`? And then consequently change it below when you pass it into the `MenuItem` component.

File: src/components/MenuItem.jsx [hunk 8]
```diff
@@ -0,0 +1,25 @@
+import React from "react";
+import { List, Image } from "semantic-ui-react";
+
+const MenuItem = ({ items }) => {
```
Comment (tochman) [inline src/components/MenuItem.jsx]: Again, this prop should be named `item` not `items`.

File: src/components/Menu.jsx [hunk 9]
```diff
@@ -1,11 +1,29 @@
-import React from 'react'
+import React, { useState, useEffect } from "react";
+import { Segment, List } from "semantic-ui-react";
+import MenuItem from "./MenuItem";
+import axios from "axios";
 
 const Menu = () => {
+  const [menuItems, setIdentification] = useState([]);
```
Comment (tochman) [inline src/components/Menu.jsx]: shouldn't `setIdentification` be named the more conventional `setMenuItems`?

File: src/components/Menu.jsx [hunk 10]
```diff
@@ -1,11 +1,29 @@
...
+  useEffect(() => {
+    axios.get("https://slowfood.heroku.com/api/products").then((response) => {
```
Comment (tochman) [inline src/components/Menu.jsx:9]: Hmm... Is `https://slowfood.heroku.com` the real URL? Also, you are intercepting a call to `localhost:3000` in your tests.

**Review by tochman — CHANGES_REQUESTED** [review]
Small stylistic and naming issues left to fix.

**Comment by tochman** [comment]
Please revisit the description and make sure it is correct. We are NOT querying a file for products, but rather an API

File: src/components/Menu.jsx (hunk 10 above)
Comment (Fraseer) [reply src/components/Menu.jsx:9]: That is not a real url, just a placeholder. localhost:3000 was removed from the test in the last commit

File: src/components/Menu.jsx (hunk 9 above)
Comment (Fraseer) [reply src/components/Menu.jsx]: Correct changed

File: src/components/Menu.jsx (hunk 10 above)
Comment (tochman) [reply src/components/Menu.jsx:9]: Okay

**Review by tochman — APPROVED** [review]
Let's go with this.

**Pull request merged**